### DISCUSSION TITLE
Database Stub Not working Using Specmatic

### DISCUSSION THREAD

**Discussion by himanshuwadekar** [specmatic/specmatic#1036] [Ask For Help] [answered]
[idproof.json](https://github.com/znsio/specmatic/files/14601300/idproof.json)
[pincode1.json](https://github.com/znsio/specmatic/files/14601301/pincode1.json)

I am encountering issues with two stubs in my project. Here are the details:

Stub 1: idproof.json
Issue: When working with idproof.json stub, I receive the following error:

Error in process method 'The database returned no natively generated identity value'.

Description: The table contains the primary key, which is also auto-incremented. However, despite this setup, the error persists.
Query: The query I'm using is:

@Query("select deletionFlag from NonServicePincodeList n where n.pincode = ?1")

Stub 2: pincode1.json
Issue: When dealing with pincode1.json stub, I encounter the following error:

Error in process method 'For input string: "false"'.

Description: The error indicates a problem with the data format. The input string "false" is causing the issue.
Method: I'm using the following method:

uploadIdProofRepository.save(uploadIDProof);

I've verified the stubs, and they seem correct. However, they are not functioning as expected. Any assistance in resolving these issues would be greatly appreciated. Thank you.

<img width="1728" alt="Screenshot 2024-03-14 at 4 51 35 PM" src="https://github.com/znsio/specmatic/assets/163411279/d4f15eb3-09a3-482a-a36d-811f25382ee7">

<img width="1728" alt="Screenshot 2024-03-14 at 4 51 43 PM" src="https://github.com/znsio/specmatic/assets/163411279/374a0ce7-ff63-4e32-abbc-83d7800da573">

<img width="1728" alt="Screenshot 2024-03-14 at 4 50 34 PM" src="https://github.com/znsio/specmatic/assets/163411279/5a2b5a7d-d349-48f0-913f-91e03037112b">

**Comment by harikrishnan83** [comment]
@himanshuwadekar thanks for reaching out. I see two errors in the screenshots you have shared
- `Query not stubbed` - This happens if the query in the idproof.json is not matching the query fired from the application. Please review the logs to see if there is any data mismatch and set up your query stub JSONs to resolve this problem. I believe this is the same issue with your pincode.json also. Kindly check.
- `The database returned no natively generated identity value` - This is related to the hibernate annotations for the id column. I am able to reproduce the same error message by setting `@GeneratedValue(strategy = GenerationType.IDENTITY)` in our [sample project](https://github.com/znsio/specmatic-jdbc-sample/blob/master/src/main/java/com/component/products/entity/Product.java). We will look into this and get back to you.

**Comment by harikrishnan83** [comment, marked as answer]
@himanshuwadekar About the `The database returned no natively generated identity value` that I mentioned above, the fix is now available as part of `specmatic-jdbc` release `0.11.2`. Example:

For a table where the id is added as part of insert query the JSON would have looked as shown below.
```
{
  "query": "insert into product (cost,name,id) values (300,'Wooden Sofa',102)",
  "response": {
    "affectedRows": 1
  }
}
```

The same query for an auto-increment based table setup will be updated as follows. Please note how the response needs to contain the generated key for id column.
```
{
  "query": "insert into product (cost,name,id) values (300,'Wooden Sofa',default)",
  "response": {
    "affectedRows": 1,
    "generatedKeys": [
      {
        "id": 102
      }
    ]
  }
}
```
